libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py

```diff
@@ -928,83 +1007,107 @@ def create_context(self, activity):
+        """
+        Creates the connector client
         :param service_url: The service URL
         :param identity: The claims identity
+        :param audience:
 
         :return: An instance of the :class:`ConnectorClient` class
         """
 
+        if not identity:
+            # This is different from C# where an exception is raised.  In this case
+            # we are creating a ClaimsIdentity to retain compatibility with this
+            # method.
+            identity = ClaimsIdentity(
+                claims={
+                    AuthenticationConstants.AUDIENCE_CLAIM: self.settings.app_id,
+                    AuthenticationConstants.APP_ID_CLAIM: self.settings.app_id,
+                },
+                is_authenticated=True,
+            )
+
+        # For requests from channel App Id is in Audience claim of JWT token. For emulator it is in AppId claim.
+        # For unauthenticated requests we have anonymous claimsIdentity provided auth is disabled.
+        # For Activities coming from Emulator AppId claim contains the Bot's AAD AppId.
+        bot_app_id = identity.claims.get(
+            AuthenticationConstants.AUDIENCE_CLAIM
+        ) or identity.claims.get(AuthenticationConstants.APP_ID_CLAIM)
+
         # Anonymous claims and non-skill claims should fall through without modifying the scope.
-        credentials = self._credentials
-
-        if identity:
-            bot_app_id_claim = identity.claims.get(
-                AuthenticationConstants.AUDIENCE_CLAIM
-            ) or identity.claims.get(AuthenticationConstants.APP_ID_CLAIM)
-
-            if bot_app_id_claim and SkillValidation.is_skill_claim(identity.claims):
-                scope = JwtTokenValidation.get_app_id_from_claims(identity.claims)
-
-                # Do nothing, if the current credentials and its scope are valid for the skill.
-                # i.e. the adapter instance is pre-configured to talk with one skill.
-                # Otherwise we will create a new instance of the AppCredentials
-                # so self._credentials.oauth_scope isn't overridden.
-                if self._credentials.oauth_scope != scope:
-                    password = await self._credential_provider.get_app_password(
-                        bot_app_id_claim
-                    )
-                    credentials = MicrosoftAppCredentials(
-                        bot_app_id_claim, password, oauth_scope=scope
-                    )
-                    if (
-                        self.settings.channel_provider
-                        and self.settings.channel_provider.is_government()
-                    ):
-                        credentials.oauth_endpoint = (
-                            GovernmentConstants.TO_CHANNEL_FROM_BOT_LOGIN_URL
-                        )
-                        credentials.oauth_scope = (
-                            GovernmentConstants.TO_CHANNEL_FROM_BOT_OAUTH_SCOPE
-                        )
+        credentials = None
```

Comment:
_get_or_create_connector_client expects a MicrosoftAppCredentials and None will throw

maybe something like:
```python
else:
            credentials = await self.__get_app_credentials("", "")
```